### PR TITLE
Update validation for Future oracle spec to allow internal oracle spec without public keys

### LINKED ISSUE #254: Oracle Spec with filter for internal oracle data should allow no public keys
# Task Overview

In order to define an oracle spec for the internal oracle
Update the validation for futures oracle specs
So that they can be defined without specifying the public key of the oracle

## Specs
- [4478](https://github.com/vegaprotocol/vega/issues/4478)

### PR DESCRIPTION
**Pull request by guoguojin**
Closes #254

### CODE REVIEW

File: commands/proposal_submission.go [hunk 1]
```diff
@@ -375,35 +379,54 @@ func checkOracleSpec(spec *oraclespb.OracleSpecConfiguration, name string) Error
...
+		return checkOracleSpecFilters(spec, name, errs)
+	} else {
+		errs.AddForProperty("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name, ErrIsRequired)
+	}
+
+	return errs
+}
+
+func isBuiltInSpec(filters []*oraclespb.Filter) bool {
+	if len(filters) == 0 || len(filters) > 1 {
```
Comment (ValentinTrinque) [inline commands/proposal_submission.go:392]: So basically `if len(filters) != 1` right ? Or I am missing something ?

**Review by ValentinTrinque — CHANGES_REQUESTED** [review]

File: commands/proposal_submission.go [hunk 2]
```diff
@@ -375,35 +379,54 @@ func checkOracleSpec(spec *oraclespb.OracleSpecConfiguration, name string) Error
...
+	if strings.HasPrefix(filters[0].Key.String(), "vegaprotocol.builtin") {
+		return true
+	}
+
+	return false
+}
+
+func checkOracleSpecFilters(spec *oraclespb.OracleSpecConfiguration, name string, errs Errors) Errors {
+	if len(spec.Filters) == 0 {
+		errs.AddForProperty("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters", ErrIsRequired)
```
Comment (ValentinTrinque) [inline commands/proposal_submission.go:405]: I think, you can return directly after this check, and thus remove the `else` branch so we can minimise indentation :)

**Review by ValentinTrinque — CHANGES_REQUESTED** [review]

File: commands/proposal_submission.go (hunk 1 above)
Comment (guoguojin) [reply commands/proposal_submission.go:392]: You are right, I was going to do a nil check on the slice and a length check, but then remembered that you can just do len on a nil slice too and missed the obvious.

File: commands/proposal_submission.go (hunk 2 above)
Comment (guoguojin) [reply commands/proposal_submission.go:405]: Yes, missed that. Thank you.

Comment (guoguojin) [reply commands/proposal_submission.go:409]: Resolved https://github.com/vegaprotocol/protos/pull/253/commits/4fc21d339ad4e84a4aaf4df70a1ff481392db2ee

File: commands/proposal_submission.go (hunk 1 above)
Comment (guoguojin) [reply commands/proposal_submission.go]: Resolved https://github.com/vegaprotocol/protos/pull/253/commits/4fc21d339ad4e84a4aaf4df70a1ff481392db2ee

File: commands/proposal_submission.go [hunk 3]
```diff
@@ -375,35 +379,60 @@ func checkOracleSpec(spec *oraclespb.OracleSpecConfiguration, name string) Error
...
+
+		return errs
+	}
+
+	for i, filter := range spec.Filters {
+		if filter.Key == nil {
+			errs.AddForProperty(fmt.Sprintf("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters.%d.key", i), ErrIsNotValid)
```
Comment (ValentinTrinque) [inline commands/proposal_submission.go:416]: **Nit-pick / non-blocker**: You can add `continue` in the `if` and remove the `else` branch.

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**